components/merkledb/src/values.rs

```diff
@@ -261,6 +261,42 @@ impl BinaryValue for [u8; HASH_SIZE] {
     }
 }
 
+/// Concatenates buffer slices, prepends each slice with its size.
+pub fn concat_buffers(buffers: &mut [Vec<u8>]) -> Vec<u8> {
+    use std::convert::TryFrom;
+
+    let mut res_buf = Vec::new();
+    for mut buf in buffers {
+        let size = u32::try_from(buf.len()).unwrap_or_else(|_| panic!("Too large buffer"));
+        res_buf.append(&mut size.to_bytes());
+        res_buf.append(&mut buf);
+    }
+    res_buf
+}
+
+/// Splits single byte buffer into vector of separate buffers according to its size
+pub fn split_buffer_into_sized_parts<'a>(
+    bytes: &'a Cow<[u8]>,
+    qty: usize,
+) -> Result<Vec<Cow<'a, [u8]>>, failure::Error> {
```

Comment:
Doesn't this method always return `Cow::Borrowed` slices? I'd suggest to change the return type to `Vec<&[u8]>`.